lib/iris/coord_systems.py

```diff
@@ -53,10 +54,18 @@ class CoordSystem(metaclass=ABCMeta):
     grid_mapping_name = None
 
     def __eq__(self, other):
-        return (
-            self.__class__ == other.__class__
-            and self.__dict__ == other.__dict__
-        )
+        if self.__class__ != other.__class__:
```

Comment:
This `__eq__` method needs some comments to explain why you are doing a partial check of keys